internal/validators/iam/iam_validator.go

```diff
@@ -185,6 +243,51 @@ func (s *IAMRuleService) ReconcileIAMPolicyRule(rule iamRule) (*types.Validation
 	return vr, nil
 }
 
+func checkSCP(iamSvc iamApi, policyDocs []v1alpha1.PolicyDocument, policySourceArn string, policySourceType string, policySourceName string) ([]string, error) {
+	var scpFailures []string
+	var ctxEntry []iamtypes.ContextEntry
+
+	// check GetContextKeysForPrincipal to do this
+	if policySourceType == "user" {
+		ctxEntry = []iamtypes.ContextEntry{
+			{
+				ContextKeyName:   ptr.Ptr("aws:username"),
+				ContextKeyType:   "string",
+				ContextKeyValues: []string{policySourceName},
+			},
+		}
+	}
+
+	for _, doc := range policyDocs {
+		for _, statement := range doc.Statements {
+
+			simulationInput := &iam.SimulatePrincipalPolicyInput{
+				ActionNames:     statement.Actions,
+				PolicySourceArn: ptr.Ptr(policySourceArn),
+				ResourceArns:    statement.Resources,
+				ContextEntries:  ctxEntry,
+			}
+
+			sim, err := iamSvc.SimulatePrincipalPolicy(context.Background(), simulationInput)
+			if err != nil {
+				return nil, err
+			}
+
+			fmt.Println("Sim eval results", sim.EvaluationResults)
+
+			for _, result := range sim.EvaluationResults {
+				if result.EvalDecision == "implicitDeny" {
```

Comment:
I think the logic here should be as follows:
```golang
if !result.OrganizationsDecisionDetail.AllowedByOrganizations && result.EvalDecision != "allowed"  {
    # append SCP failure
}
```
Note: The above only applies if we are NOT deprecating the original implementation for non-SCP evaluation. Otherwise the logic changes and you need to delete a bunch of stuff, etc.

Reply:
Additionally, you need to handle pagination for the policy simulation results by checking `IsTruncated`.